Zend/zend_object_handlers.c

```diff
@@ -815,6 +975,7 @@ ZEND_API zval *zend_std_write_property(zend_object *zobj, zend_string *name, zva
 
 	property_offset = zend_get_property_offset(zobj->ce, name, (zobj->ce->__set != NULL), cache_slot, &prop_info);
 
+try_again:
 	if (EXPECTED(IS_VALID_PROPERTY_OFFSET(property_offset))) {
```

Comment:
```suggestion
	if (EXPECTED(IS_VALID_PROPERTY_OFFSET(property_offset))) {
try_again:
```
Is probably better - like retry only ever happens for valid offsets anyway.